File: android/app/src/main/java/bagus2x/sosmed/domain/model/Comment.kt
package bagus2x.sosmed.domain.model

import java.time.LocalDateTime

data class Comment(
    val id: Long,
    val feedId: Long,
    val parentId: Long?,
    val path: String,
    val author: Profile,
    val description: String,
    val medias: List<Media>,
    val pinned: Boolean,
    val favorite: Boolean,
    val totalFavorites: Int,
    val totalReplies: Int,
    val totalLoadedReplies: Int,
    val createdAt: LocalDateTime
) {
    val pathSize: Int
        get() = path.split(".").size
}

File: backend/models/comment.go
package models

import (
	"gopkg.in/guregu/null.v4"
	"time"
)

type Comment struct {
	Id             int64
	FeedId         int64
	ParentId       null.Int
	Path           string
	Author         Profile
	Description    string
	Medias         []Media
	TotalFavorites int
	TotalReplies   int
	CreatedAt      time.Time
}

type CommentRes struct {
	Id             int64    `json:"id"`
	FeedId         int64    `json:"feedId"`
	ParentId       null.Int `json:"parentId"`
	Path           string   `json:"path"`
	Author         Profile  `json:"author"`
	Description    string   `json:"description"`
	Medias         []Media  `json:"medias"`
	TotalFavorites int      `json:"totalFavorites"`
	TotalReplies   int      `json:"totalReplies"`
	CreatedAt      int64    `json:"createdAt"`
}

type CreateCommentReq struct {
	FeedId      int64   `json:"feedId"`
	ParentId    int64   `json:"parentId"`
	Description string  `json:"description"`
	Medias      []Media `json:"medias"`
}
